FF1Blazorizer/wwwroot/js/site.js

```diff
@@ -1,168 +1,207 @@
+                navigator.serviceWorker.register('/service-worker.js').then(reg => {
+                        console.debug('service worker registered');
+                        reg.addEventListener('updatefound', () => {
+                                console.debug('New update found');
+                                newWorker = reg.installing;
+                                newWorker.addEventListener('statechange', () => {
+                                        if (newWorker.state === 'installed' && navigator.serviceWorker.controller) {
+                                                console.debug('Showing update notification');
+                                                DotNet.invokeMethod('FF1Blazorizer', 'ShowUpdateNotification');
+                                        }
+                                });
+                        });
+                });
+
+                let refreshing;
+                navigator.serviceWorker.addEventListener('controllerchange', function () {
+                        if (refreshing) return;
+                        window.location.reload();
+                        refreshing = true;
+                });
+        }
 })
 
 
 /**
- * Call from Blazor to register the new service worker 
+ * Call from Blazor to register the new service worker
  */
 function updateServiceWorkerNow() {
-	if (newWorker) {
-		newWorker.postMessage({ action: 'skipWaiting' });
-	} else {
-		window.location.reload();
-	}
+        if (newWorker) {
+                newWorker.postMessage({ action: 'skipWaiting' });
+        } else {
+                window.location.reload();
+        }
 }
 
 let pwa;
-window.addEventListener('beforeinstallprompt', (e) => {	
-	e.preventDefault();
-	pwa = e;
+window.addEventListener('beforeinstallprompt', (e) => { 
+        e.preventDefault();
+        pwa = e;
 });
 
 function showPWAInstall() {
-	pwa?.prompt();
-	pwa = null;
+        pwa?.prompt();
+        pwa = null;
 }
+
+function setFFRPreferences(keyname, prefdata) {
+    var iframe = document.getElementsByTagName('iframe')[0];
+    var win;
+    // some browser (don't remember which one) throw exception when you try to access
+    // contentWindow for the first time, it work when you do that second time
+    try {
+        win = iframe.contentWindow;
+    } catch(e) {
+        win = iframe.contentWindow;
+    }
+    // save obj in subdomain localStorage
+    win.postMessage(JSON.stringify({key: keyname, method: "set", data: prefdata}), "*");
+};
+
+window.FFRPreferencesCallbacks = {};
+window.onmessage = function(e) {
+    //if (e.origin != "http://other.example.com") {
+    //    return;
+    //}
+    var response = JSON.parse(e.data);
+    if (window.FFRPreferencesCallbacks[response.key]) {
+        var resolve = window.FFRPreferencesCallbacks[response.key];
+        resolve(response.data);
+    }
+};
+
+async function getFFRPreferences(keyname) {
+    var iframe = document.getElementsByTagName('iframe')[0];
```

Comment:
use an id and grab it by that instead, so this doenst break if another iframe is added before this one.

Reply:
Done